cmd/lakectl/cmd/root.go

```diff
@@ -208,6 +213,52 @@ func getSyncArgs(args []string, requireRemote bool, considerGitRoot bool) (remot
 	return
 }
 
+func withMessageFlags(cmd *cobra.Command, defaultMsg string, allowEmpty bool) {
+	cmd.Flags().StringP(commitMsgFlagName, "m", defaultMsg, "commit message")
+	cmd.Flags().Bool(allowEmptyMsgFlagName, allowEmpty, "allow an empty commit message")
+}
+
+func withMetadataFlag(cmd *cobra.Command) {
+	cmd.Flags().StringSlice(metaFlagName, []string{}, "key value pair in the form of key=value")
+}
+
+func withCommitFlags(cmd *cobra.Command, defaultMsg string, allowEmptyMessage bool) {
+	withMessageFlags(cmd, defaultMsg, allowEmptyMessage)
+	withMetadataFlag(cmd)
+}
+
+func getCommitFlags(cmd *cobra.Command) (string, map[string]string) {
+	message := Must(cmd.Flags().GetString(commitMsgFlagName))
+	emptyMessageBool := Must(cmd.Flags().GetBool(allowEmptyMsgFlagName))
+	if strings.TrimSpace(message) == "" && !emptyMessageBool {
+		DieFmt(fmtErrEmptyMsg)
+	}
+
+	kvPairs, err := getKV(cmd, metaFlagName)
+	if err != nil {
+		DieErr(err)
+	}
+
+	return message, kvPairs
+}
+
+func getKV(cmd *cobra.Command, name string) (map[string]string, error) { //nolint:unparam
+	kvList, err := cmd.Flags().GetStringSlice(name)
+	if err != nil {
+		return nil, err
+	}
+
+	kv := make(map[string]string)
```

Comment:
Is nil different than an empty map? I want to make sure we're not going to show an empty map (`{}`) everywhere in the UI, CLI, etc.

Reply:
The code did not change - just moved to a new file. The behavior remains the same as in commit today